# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: intl/l10n/Localization.cpp
# Commit: 626b40e3aad1
# Full Hash: 626b40e3aad147d59a439b0bd1d5dbd328ad6b1a
# Author: Zibi Braniecki <zbraniecki@mozilla.com>
# Date: 2021-08-02 21:43:56
# Regressor Bug: 1613705
# File Overlap Count: 1
# Description:
#   Bug 1613705 - [localization] part9: Update DOMLocalization to match Localization constructor. r=nika,emilio
#   
#   Depends on D113691
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D113692
# ==============================================================================

diff -r 1cbab987bb7d -r 626b40e3aad1 intl/l10n/Localization.cpp
--- a/intl/l10n/Localization.cpp	Mon Aug 02 09:54:10 2021 +0000
+++ b/intl/l10n/Localization.cpp	Mon Aug 02 09:54:10 2021 +0000
@@ -237,31 +237,21 @@
 
 void Localization::OnChange() { ffi::localization_on_change(mRaw.get()); }
 
-void Localization::AddResourceId(const nsAString& aResourceId) {
-  NS_ConvertUTF16toUTF8 resId(aResourceId);
-  ffi::localization_add_res_id(mRaw.get(), &resId);
+void Localization::AddResourceId(const nsACString& aResourceId) {
+  ffi::localization_add_res_id(mRaw.get(), &aResourceId);
 }
 
-uint32_t Localization::RemoveResourceId(const nsAString& aResourceId) {
-  NS_ConvertUTF16toUTF8 resId(aResourceId);
-  return ffi::localization_remove_res_id(mRaw.get(), &resId);
+uint32_t Localization::RemoveResourceId(const nsACString& aResourceId) {
+  return ffi::localization_remove_res_id(mRaw.get(), &aResourceId);
 }
 
-void Localization::AddResourceIds(const nsTArray<nsString>& aResourceIds) {
-  nsTArray<nsCString> resIds(aResourceIds.Length());
-  for (const auto& resId : aResourceIds) {
-    resIds.AppendElement(NS_ConvertUTF16toUTF8(resId));
-  }
-  ffi::localization_add_res_ids(mRaw.get(), &resIds);
+void Localization::AddResourceIds(const nsTArray<nsCString>& aResourceIds) {
+  ffi::localization_add_res_ids(mRaw.get(), &aResourceIds);
 }
 
 uint32_t Localization::RemoveResourceIds(
-    const nsTArray<nsString>& aResourceIds) {
-  nsTArray<nsCString> resIds(aResourceIds.Length());
-  for (const auto& resId : aResourceIds) {
-    resIds.AppendElement(NS_ConvertUTF16toUTF8(resId));
-  }
-  return ffi::localization_remove_res_ids(mRaw.get(), &resIds);
+    const nsTArray<nsCString>& aResourceIds) {
+  return ffi::localization_remove_res_ids(mRaw.get(), &aResourceIds);
 }
 
 already_AddRefed<Promise> Localization::FormatValue(